src/core_ocean/mode_forward/mpas_ocn_time_integration_split.F

```diff
@@ -794,10 +799,24 @@ subroutine ocn_time_integrator_split(domain, dt)!{{{
                                               uPerpTime)
                      call mpas_pool_get_array(statePool, 'normalBarotropicVelocitySubcycle', normalBarotropicVelocitySubcycleNew, &
                                               newBtrSubcycleTime)
-                     call mpas_pool_get_array(statePool, 'sshSubcycle', sshSubcycleCur, oldBtrSubcycleTime)
 
                      call mpas_pool_get_array(diagnosticsPool, 'barotropicForcing', barotropicForcing)
 
+                     ! Subtract tidal potential from ssh, if needed
+                     if (config_use_tidal_potential_forcing) then
+                       call mpas_pool_get_array(statePool, 'sshSubcycle', sshSubcycleCurNoTides, oldBtrSubcycleTime)
+                       call mpas_pool_get_array(forcingPool, 'sshSubcycleCurWithTides', sshSubcycleCur)
+                       call mpas_pool_get_array(forcingPool, 'tidalPotentialEta', tidalPotentialEta)
+                       call mpas_pool_get_dimension(block % dimensions, 'nCells', nCellsPtr)
+
+                       nCells = nCellsPtr
+                       do iCell = 1, nCells
+                         sshSubcycleCur(iCell) = sshSubcycleCurNoTides(iCell) - tidalPotentialEta(iCell)
```

Comment:
@xylar, if `config_use_tidal_potential_forcing` is true, the `sshSubcycleCur` variable name is aliased to the temporary array `sshSubcycleCurWithTides` and 'sshSubcycleCurNoTides' is assigned to the actual SSH. This way, the SSH is not modified. I did this so the subcycle computation could remain unchanged and that no additional calculations would be done there.

I agree it's a little confusing, but @mark-petersen was (rightly) concerned about adding FLOPS to the subcycle loop for a feature that is not going to be used for E3SM at this point. This is what I came up with to get around that. I'm definitely open to other suggestions.

Reply:
@sbrus89, my small concern is the naming convention for the variables in the code being inconsistent with the names in the pools.  I'd change that if I were you but it's not critical.

My big concern is the implementation of tides as a modification to SSH.  I do not believe that is a correct approach, and they should instead be a modification of the gravitational potential, which would result in the modification of the SSH as an emergent property (as @pwolfram says above).  The design doc gives the incorrect impression that modifying SSH is the way to go because this is equivalent to modifying the gravitational potential for a shallow water model.  But the same is not true of a 3D, primitive-equation ocean model.  Thus, my comment pointing to the 2 chapters of the MOM5 manual that discuss modification of the gravitational potential in general and then specifically for tides.  I think that's the (only?) way to go.

Reply:
Talking with @pwolfram on the phone, I got assurances that @mark-petersen and he would verify that these modifications to the subcycled SSH do not feed into the SSH at the end of a time step, and are just used to compute the gravitational potential in the barotropic subcycle.

Reply:
Since each of the subcycle calculations is being done inside blocks, the pointer association to the working array that contains ssh-tidal_potential will not effect anything outside of that particular block.